projects/greedy_infomax/experiments/default_base.py

```diff
@@ -0,0 +1,256 @@
+)
+from nupic.research.frameworks.vernon.distributed import experiments, mixins
+
+
+class GreedyInfoMaxExperiment(
+    mixins.LogEveryLoss,
+    mixins.NoiseRobustnessTest,
+    experiments.SelfSupervisedExperiment,
+):
+    def transform_data_to_device_unsupervised(self, data, target, device, non_blocking):
+        data = data.to(self.device, non_blocking=non_blocking)
+        target = target.to(self.device, non_blocking=non_blocking)
+        return data, target
+
+    def create_loaders(self, config):
+        unsupervised_data = self.load_dataset(config, dataset_type="unsupervised")
+        if config.get("reuse_unsupervised_dataset", False):
+            supervised_data = unsupervised_data
+        else:
+            supervised_data = self.load_dataset(config, dataset_type="supervised")
+        validation_data = self.load_dataset(config, dataset_type="validation")
+        num_unsupervised_samples = config.get("num_unsupervised_samples", -1)
+        if num_unsupervised_samples > 0:
+            unsupervised_indices = np.random.choice(
+                len(unsupervised_data), num_unsupervised_samples, replace=False
+            )
+            unsupervised_data = Subset(unsupervised_data, unsupervised_indices)
+
+        num_supervised_samples = config.get("num_supervised_samples", -1)
+        if num_supervised_samples > 0:
+            supervised_indices = np.random.choice(
+                len(supervised_data), num_supervised_samples, replace=False
+            )
+            supervised_data = Subset(supervised_data, supervised_indices)
+
+        num_validation_samples = config.get("num_validation_samples", -1)
+        if num_validation_samples > 0:
+            validation_indices = np.random.choice(
+                len(validation_data), num_validation_samples, replace=False
+            )
+            validation_data = Subset(validation_data, validation_indices)
+
+        self.unsupervised_loader = (
+            self.train_loader
+        ) = self.create_unsupervised_dataloader(config, unsupervised_data)
+
+        self.supervised_loader = self.create_supervised_dataloader(
+            config, supervised_data
+        )
+        self.val_loader = self.create_validation_dataloader(config, validation_data)
+
+    # avoid changing key names for sigopt
+    @classmethod
+    def get_readable_result(cls, result):
+        return result
+
+
+BATCH_SIZE = 32
+NUM_CLASSES = 10
+NUM_EPOCHS = 60
+DEFAULT_BASE = dict(
+    experiment_class=GreedyInfoMaxExperiment,
+    # wandb
+    wandb_args=dict(
+        project="greedy_infomax-replication", name="s20/default-lr/full-epoch/"
+    ),
+    # Dataset
+    dataset_class=STL10,
+    dataset_args=STL10_DATASET_ARGS,
+    #               STL10:
+    # 500 training images (10 pre-defined folds)
+    # 8000 test images (800 test images per class)
+    # 100,000 unlabeled images
+    # num_unsupervised_samples=1000,
+    # num_supervised_samples=500,
+    # num_validation_samples=32,
```

Comment:
In this config, you have a lot of lines of code which are commented out. You should delete these if they're not used.

Reply:
Removed commented lines